packages/remix-dev/vite/plugin.ts

```diff
@@ -1238,27 +1254,8 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
           let ssrViteManifest = await loadViteManifest(serverBuildDirectory);
           let clientViteManifest = await loadViteManifest(clientBuildDirectory);
 
-          let clientAssetPaths = new Set(
-            Object.values(clientViteManifest).flatMap((chunk) => [
-              ...(chunk.css ?? []),
-              ...(chunk.assets ?? []),
-            ])
-          );
-
-          // Handle `.css?url` files that only exist in SSR module graph
-          let ssrCssUrlFilePaths = Object.values(ssrViteManifest)
-            .filter((chunk) => chunk.file.endsWith(".css"))
-            .map((chunk) => chunk.file);
-
-          // Handle generic assets that only exist in SSR module graph
-          let ssrChunkAssetPaths = Object.values(ssrViteManifest).flatMap(
-            (chunk) => chunk.assets ?? []
-          );
-
-          let ssrAssetPaths = new Set<string>([
-            ...ssrCssUrlFilePaths,
-            ...ssrChunkAssetPaths,
-          ]);
+          let clientAssetPaths = getViteManifestAssetPaths(clientViteManifest);
```

Comment:
This was the root of the issue. The logic for extracting assets from the client and server builds differed. These need to be the same because we compare their assets to detect server-only assets that need to be moved to the client build directory. The linked issue was because we were attempting to move a file to a path that already existed.